src/werkzeug/sansio/utils.py

```diff
@@ -91,3 +93,60 @@ def get_host(
         if not host_is_trusted(host, trusted_hosts):
             raise SecurityError(f'Host "{host}" is not trusted')
     return host
+
+
+def get_current_url(
+    scheme: str,
+    host: str,
+    path: str,
+    root_path: str,
+    query_string: bytes,
+    root_only: bool = False,
```

Comment:
Having these parameters isn't really necessary, at least in any of the places this function is used. We could just as easily make the other parameters default to `None`, and stop building when we reach the first `None`.